datafusion/src/scalar.rs

```diff
@@ -333,6 +333,29 @@ impl std::hash::Hash for ScalarValue {
     }
 }
 
+impl num::ToPrimitive for ScalarValue {
+    fn to_i64(&self) -> Option<i64> {
+        use ScalarValue::*;
+        match self {
+            Int8(v) => Some(v.unwrap() as i64),
+            Int16(v) => Some(v.unwrap() as i64),
+            Int32(v) => Some(v.unwrap() as i64),
+            Int64(v) => Some(v.unwrap() as i64),
+            _ => None,
+        }
+    }
+    fn to_u64(&self) -> Option<u64> {
+        use ScalarValue::*;
+        match self {
+            UInt8(v) => Some(v.unwrap() as u64),
```

Comment:
It might be ok to  implement for `Int8` and the other signed types as long as the value is greater than 0